dipy/tracking/local/localtrack.pyx

```diff
@@ -208,3 +210,263 @@ cdef int _local_tracker(
+                        particle_weights[0, p] > 0):
+                    particle_states[0, p, 0] = TRACKPOINT
+
+        sum_weights = 0
+        for p in range(particle_count):
+            sum_weights += particle_weights[0, p]
+        sum_squared = 0
+        for p in range(particle_count):
+            particle_weights[0, p] = particle_weights[0, p] / sum_weights
+            sum_squared += particle_weights[0, p] * particle_weights[0, p]
+
+        # Resample the particles if the weights are too uneven.
+        # Particles with negligable weights are replaced by duplicates of
+        # those with high weigths through resamplingip
+        N_effective = 1. / sum_squared
+        if N_effective < particle_count / 10.:
+            # copy data in the temp arrays
+            for pp in range(particle_count):
+                for ss in range(pft_nbr_steps):
+                    for j in range(3):
+                        particle_paths[1, pp, ss,
+                                       j] = particle_paths[0, pp, ss, j]
+                        particle_dirs[1, pp, ss,
+                                      j] = particle_dirs[0, pp, ss, j]
+                particle_weights[1, pp] = particle_weights[0, pp]
+                particle_states[1, pp, 0] = particle_states[0, pp, 0]
+                particle_states[1, pp, 1] = particle_states[0, pp, 1]
+
+            # sample N new particle
+            cumsum(& particle_weights[1, 0],
+                   & particle_weights[1, 0],
+                   particle_count)
+            for pp in range(particle_count):
+                rdm_sample = random() * particle_weights[1, particle_count - 1]
+                p_source = where_to_insert(& particle_weights[1, 0],
+                                           rdm_sample,
+                                           particle_count)
+                for ss in range(pft_nbr_steps):
+                    for j in range(3):
+                        particle_paths[0, pp, ss,
+                                       j] = particle_paths[1, p_source, ss, j]
+                        particle_dirs[0, pp, ss,
+                                      j] = particle_dirs[1, p_source, ss, j]
+                particle_states[0, pp, 0] = particle_states[1, p_source, 0]
+                particle_states[0, pp, 1] = particle_states[1, p_source, 1]
+                particle_weights[0, pp] = 1. / particle_count
+
+    # update the streamline with the trajectory of one particle
+    cumsum(& particle_weights[0, 0],
+           & particle_weights[0, 0],
+           particle_count)
+    rdm_sample = random() * particle_weights[0, particle_count - 1]
+    p = where_to_insert(& particle_weights[0, 0], rdm_sample, particle_count)
+
+    for s in range(particle_states[0, p, 1]):
+        for j in range(3):
+            streamline[streamline_i + s, j] = particle_paths[0, p, s, j]
+            directions[streamline_i + s, j] = particle_dirs[0, p, s, j]
+
+    ###TODO fix this
```

Comment:
@nilgoyette, do you know why cython doesn't let me do `tissue_class[0] = TissueClass(particle_states[0, p, 0])`
```
NameError: name 'TissueClass' is not defined
```
I do have
```
from .tissue_classifier cimport TissueClass
```

Reply:
It seems that a cython enum doesn't "exists". It's simply defining the constants in the enum. You can cast it to the right type though:

    tissue_class[0] = <TissueClass>particle_states[0, p, 0]

I tested and it compiles and all the tests pass. There's no way to check if the `particle_states` returns a valid `TissueClass` but hey, it's cython!